src/components/MessageBox/MessageBox.spec.js

```diff
@@ -1,11 +1,18 @@
 import '@testing-library/jest-dom';
 import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 import MessageBox from './index';
 
-describe('MessageBox component', () => {
+const mockedUsedNavigate = jest.fn();
+jest.mock('react-router-dom', () => ({
+	...jest.requireActual('react-router-dom'),
+	useNavigate: () => mockedUsedNavigate,
+}));
+
+describe('MessageBox component with type "delete"', () => {
 	it('should render children', () => {
-		render(<MessageBox text="hey" />);
+		render(<MessageBox type="delete" text="hey" />);
 		const msg = screen.getByText('hey');
 		const cancelBTN = screen.getByText('cancel');
 		const okBTN = screen.getByText('ok');
```

Comment:
I would write the variables out. It would make them much more readable. Same here with the buttons as well though:
```suggestion
		const message = screen.getByText('hey');
		const cancelButton = screen.getByRole('button', {name: 'cancel'});
		const okButton = screen.getByRole('button', {name: 'ok'});
```

Reply:
I try it, but I don't get the buttons by their roles.

Reply:
I think because you overwrite the roles of the button in the `MessageBox` component:
```jsx
						<Button
							variant="medium"
							role="cancel"
							onClick={() => {
								setModal('', '', '');
								setModalStatus(false);
							}}
						>
							cancel
						</Button>
```
I would recommend to not use `role` here.